### DISCUSSION TITLE
Linking DNS Zones with Contacts, and maybe Sites

### DISCUSSION THREAD

**Discussion by dywilson73** [sys4/netbox-plugin-dns#71] [Ideas]
Some of this might be specific to our implementation, but here goes. 

We are managing a lot of DNS Records for customers and we would like to be able to assign a contact to any given domain Zone. This allows us to then automate notifications, and reminders related to any DNS entry / Zone. 

More background - We found that netbox is great for allocating different types of customer contacts to any given infrastructure, device, virtual machine and adding domains would be helpful. We use billing contacts, technical contacts and Notification contacts. It would be great to be able to automatically issue notifications to contacts for various circumstances relating to their Domain. 

We also often assign sites to customers, we then assign devices, virtual machines and applications to the site to tie them to a customer, and then we allocate billing, technical and notification contacts to the site. 

Assigning a Zone to a site just makes it easier to manage changes to contacts, changing contacts on a site would be done once for all netbox objects linked to that site.

**Comment by peteeckel** [comment]
Hi @dywilson73, thanks for your input!

What you are describing in the first part of your post is pretty much the use case for tenancy, which NetBox DNS supports starting with version 0.19.0.

You assign Tenant objects to customers and Contact objects for contacts within their organisation to a Tenant, and then assign a Tenant to a Zone - consistent with how NetBox models it for other objects.

Regarding Site objects I don't see a generic use case for adding them to NetBox DNS. But you're not out of options for this specific requirement either: If you have Site objects in NetBox, you can always add a custom field of that type to your Zone in NetBox DNS and use it in whatever way you need to. Same with Contact objects if you think the tenancy feature is oversized for your needs, by the way.

Does that make sense to you?

**Reply by dywilson73** [reply]
Thanks, 

I am still running 18.2.0 and so did not see that as an option. 
Tenancy could work, I will have to re-look at it - but on face value we would pretty much have one tennant for each and every site. 
So the tenancy option has not been attractive, but some of the other benefits of tennancy are likely to push us ocross that line anyway.

Looks like ill need to do some updates (core and plugins) and then take another look. 

Thanks
